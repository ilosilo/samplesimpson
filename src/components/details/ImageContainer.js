import * as React from 'react';
import {
    View,
    StyleSheet,
    Image,
    Text
} from 'react-native';

const ImageContainer = (props) => {
    return (
        <View style={st.imageContainerStyle} >
            {props.item.avatar? 
            <Image
                style={{ width: 200, height: 200 }}
                source={{ uri: props.item.avatar }}
            /> : 
            <Text style={st.noImageText}> Image Not Found </Text>}
        </View>
    );
};
export default ImageContainer;

const st = StyleSheet.create({
    imageContainerStyle: {
        width: '100%',
        height: 220,
        margin: 20,
        justifyContent: 'center',
        alignItems: 'center',
        alignContent: 'center'
    },
    noImageText:{
        color:'gray',
        fontStyle:'italic',
        fontSize:14,
        alignSelf:'center'
    }

});
