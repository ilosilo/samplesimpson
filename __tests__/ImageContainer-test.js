
 import 'react-native';
 import React from 'react';
 import ImageContainer from '../src/components/details/ImageContainer';
 import renderer from 'react-test-renderer';
 
 test('renders correctly', () => {
     const item = {"id":"3","name":"Bart Simpson","avatar":"https://static.wikia.nocookie.net/simpsons/images/6/65/Bart_Simpson.png/revision/latest/scale-to-width-down/250?cb=20190409004756","job":"Student","about":"Bartholomew JoJo \"Bart\" Simpson "}
    const tree = renderer.create(<ImageContainer item={item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });