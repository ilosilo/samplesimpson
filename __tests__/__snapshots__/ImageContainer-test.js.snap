// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "height": 220,
      "justifyContent": "center",
      "margin": 20,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://static.wikia.nocookie.net/simpsons/images/6/65/Bart_Simpson.png/revision/latest/scale-to-width-down/250?cb=20190409004756",
      }
    }
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
</View>
`;
